quickjs.c

```diff
@@ -43787,10 +43809,18 @@ static JSMapRecord *map_add_record(JSContext *ctx, JSMapState *s,
     mr->map = s;
     mr->empty = FALSE;
     if (s->is_weak) {
-        pmr = get_first_weak_ref(key);
+        JSWeakRefRecord *wr, **pwr;
+        wr = js_malloc(ctx, sizeof(*wr));
```

Comment:
Would it be possible to embed the JSWeakRefRecord inside JSMapRecord so you don't have to do this allocation? That's one fewer failure state.

If you're worried about embiggening JSMapRecord with three words, you could drop the `kind` field and use the LSB of the map_record/weak_ref_data pointer to distinguish between them.

I suppose the counterargument to my suggestion is that WeakMap and WeakRef aren't very common and therefore not worth optimizing to death. :)

Reply:
I'm taking the way out on this one :-P